streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -17,18 +17,24 @@
 package org.apache.kafka.streams.processor.internals;
 
 import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.utils.LogContext;
 import org.apache.kafka.streams.processor.StateStore;
 import org.apache.kafka.streams.processor.TaskId;
 
 import java.util.Collection;
 import java.util.Set;
+import org.slf4j.Logger;
 
+import static java.lang.String.format;
 import static org.apache.kafka.streams.processor.internals.Task.State.CLOSED;
 import static org.apache.kafka.streams.processor.internals.Task.State.CREATED;
 
 public abstract class AbstractTask implements Task {
     private Task.State state = CREATED;
 
+    protected final Logger log;
```

Comment:
Pulled this up into `AbstractTask` so we can log in the `executeAndMaybeSwallow` method I also extracted

Reply:
Hmmm... Seems to not align with OO principles to pass in the task type... maybe better pass the `Logger` into `executeAndMaybeSwallow` ?

Reply:
Ack